core/domain/suggestion_registry.py

```diff
@@ -156,11 +157,27 @@ def validate(self):
                 'Expected author_id to be a string, received %s' % type(
                     self.author_id))
 
-        if not isinstance(self.final_reviewer_id, python_utils.BASESTRING):
-            if self.final_reviewer_id:
+        if (
+                self.author_id is not None and
+                not user_id_migration.verify_user_id_correct(self.author_id)
```

Comment:
Don't add this service layer checks here, should we move this to suggestion_service before creating and saving suggestions?

Reply:
These checks will be made in suggestion_service before creating suggestions, as the domain object is created and its `validate` method is called before modifying the datastore.